lib/dio.dart

```diff
@@ -0,0 +1,94 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:dio/dio.dart';
+import 'package:fullstory_flutter/fs.dart';
+
+/// Captures requests and responses from dio for display in Fullstory.
+class FSInterceptor extends Interceptor {
+  final _requestStartTimes = <Uri, int>{};
+
+  /// Computes the size (in bytes) of request data.
+  ///
+  /// Defaults to compute the size of the data's toString result in UTF8.
+  final int Function(RequestOptions) computeRequestSize;
+
+  /// Computes the size (in bytes) of response data.
+  ///
+  /// Defaults to compute the size of the data's toString result in UTF8.
+  final int Function(Response?) computeResponseSize;
+
+  /// Creates a [FSInterceptor] that captures network events.
+  ///
+  /// The [computeRequestSize] and [computeResponseSize] functions are used to
+  /// compute the size of the request and response data. By default, they
+  /// assime UTF-8 encoding. If your request uses another enocoding, override
+  /// these fields.
+  FSInterceptor({
+    this.computeRequestSize = requestFromUtf8,
+    this.computeResponseSize = responseFromUtf8,
```

Comment:
Could we default to using the `content-length` header when it's present? (Or do we need the decompressed size?)

Also, it looks like it also handles `Uint8List`, although I suppose that's generally the same thing.